From 4a7de49de2ad3de001d712b6cb020a5b8e65a1e1 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Tue, 24 Jul 2018 16:32:41 +0300
Subject: [PATCH 4247/5242] MLK-18809: drm: imx: dcss: revert the max upscale
 ratio to 1:8

commit  a4dadbca376f2d6e88103a7027b228763c3ddd4d from
https://source.codeaurora.org/external/imx/linux-imx.git

The DCSS RM states that the maximum upscale ratio is 1:8. However, DCSS
HW team suggested some time back that upscale ratios of 1:16 could be
achieved with DCSS scaler, though these ratios were not validated.

Unfortunately there are corner cases, when the upscale ratio nears
1:16, that fail. At the recommendation of DCSS designers, we revert the
maximum ratio to 1:8 until further notice.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
(cherry picked from commit bafb5a9481c289951872923d4bbbbcf24a8910b5)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-scaler.c |   12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/gpu/imx/dcss/dcss-scaler.c b/drivers/gpu/imx/dcss/dcss-scaler.c
index 2ca9c04..84e6859 100644
--- a/drivers/gpu/imx/dcss/dcss-scaler.c
+++ b/drivers/gpu/imx/dcss/dcss-scaler.c
@@ -522,15 +522,15 @@ struct dcss_scaler_ratios {
 };
 
 static const struct dcss_scaler_ratios dcss_scaler_ratios[] = {
-	{max_downscale(3), max_upscale(16)},
-	{max_downscale(5), max_upscale(16)},
-	{max_downscale(5), max_upscale(16)},
+	{max_downscale(3), max_upscale(8)},
+	{max_downscale(5), max_upscale(8)},
+	{max_downscale(5), max_upscale(8)},
 };
 
 static const struct dcss_scaler_ratios dcss_scaler_wrscl_ratios[] = {
-	{max_downscale(5), max_upscale(16)},
-	{max_downscale(7), max_upscale(16)},
-	{max_downscale(7), max_upscale(16)},
+	{max_downscale(5), max_upscale(8)},
+	{max_downscale(7), max_upscale(8)},
+	{max_downscale(7), max_upscale(8)},
 };
 
 static bool dcss_scaler_fractions_set(struct dcss_soc *dcss, int ch_num,
-- 
1.7.9.5

